### ISSUE TITLE
readout_format fails silently

### ISSUE THREAD

**Issue by adwasser** [jupyter-widgets/ipywidgets#1544]
IPython version 6.1.0

Any improper string input for the `readout_format` keyword argument of `FloatSlider` is associated with the interactive failing to display, without any indication that the `readout_format` input is bad.  

Is there a way to warn users about this?

![widgets](https://user-images.githubusercontent.com/4566243/28648049-d583a4d2-721f-11e7-892c-82407261acc1.png)

**Comment by pbugnion** [comment]
Thanks for raising an issue. For reference, this is still broken on current master. The view throws when it renders.

For a minimal example:

```python
import ipywidgets as widgets
widgets.FloatSlider(1.0, readout_format='broken')
```

[This](https://github.com/jupyter-widgets/ipywidgets/blob/master/packages/controls/src/widget_float.ts#L66) is the line that causes the exception. 

@jasongrout Do you think the solution to this is to define a more specific traitlet for the readout format, or to somehow expose the exception thrown by the JS client in the Python layer?

**Comment by jasongrout** [comment]
> define a more specific traitlet for the readout format

I like this more than trying to push errors back to python. Is there a nice regex for these formats? What if we had a string traitlet that validated based on a regex?

**Comment by jasongrout** [comment]
(Marking as Future, but if someone submits a PR in time for 7.0, that's great too)

**Comment by pbugnion** [comment]
> What if we had a string traitlet that validated based on a regex?

[This](https://github.com/d3/d3-format/blob/master/src/formatSpecifier.js) seems useful. The bulk of the parsing is done by a regex.

> Marking as Future, but if someone submits a PR in time for 7.0, that's great too

Happy to take a look at this over the weekend, unless someone else gets there first.

**Comment by jasongrout** [comment]
Fixed in #1550.
